src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,100 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const resetAll = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const areAllRequiredFieldsFilled = (
+    title
+    && imgUrl
+    && imdbUrl
+    && imdbId
+    && [title, imgUrl, imdbUrl, imdbId].every(
+      field => field.trim() !== '',
+    ));
```

Comment:
you can remove all the previous checks, you basically do the same in the `every` call
```suggestion
   [title, imgUrl, imdbUrl, imdbId].every(
      field => field.trim() !== '',
    ));
```